README.md

```diff
@@ -665,6 +680,59 @@ variables := map[string]interface{}{
 }
 ```
 
+### Debugging and Unit test
+
+Enable debug mode with the `WithDebug` function. If the request is failed, the request and response information will be included in `extensions[].internal` property.
+
+```json
+{
+	"errors": [
+		{
+			"message":"Field 'user' is missing required arguments: login",
+			"extensions": {
+				"internal": {
+					"request": {
+						"body":"{\"query\":\"{user{name}}\"}",
+						"headers": {
+							"Content-Type": ["application/json"]
+						}
+					},
+					"response": {
+						"body":"{\"errors\": [{\"message\": \"Field 'user' is missing required arguments: login\",\"locations\": [{\"line\": 7,\"column\": 3}]}]}",
+						"headers": {
+							"Content-Type": ["application/json"]
+						}
+					}
+				}
+			},
+			"locations": [
+				{
+					"line":7,
+					"column":3
+				}
+			]
+		}
+	]
+}
+```
+
+Because the GraphQL query string is generated in runtime using reflection, it isn't really safe. To assure the GraphQL query is expected, it's necessary to write some unit test for query construction.
+
+```go
+// ConstructQuery build GraphQL query string from struct and variables
+func ConstructQuery(v interface{}, variables map[string]interface{}, options ...Option) (string, error)
+
+// ConstructQuery build GraphQL mutation string from struct and variables
+func ConstructMutation(v interface{}, variables map[string]interface{}, options ...Option) (string, error)
+
+// ConstructSubscription build GraphQL subscription string from struct and variables
+func ConstructSubscription(v interface{}, variables map[string]interface{}, options ...Option) (string, error) 
+
+// UnmarshalGraphQL parses the JSON-encoded GraphQL response data and stores
+// the result in the GraphQL query data structure pointed to by v.
+func UnmarshalGraphQL(data []byte, v interface{}) error 
+```
```

Comment:
The functions above should help us in writing unit tests on the GraphQL data types, without having to actually make the query.